processor/src/fast/mod.rs

```diff
@@ -0,0 +1,881 @@
+use alloc::vec::Vec;
+use core::cmp::min;
+
+use memory::Memory;
+use miden_air::RowIndex;
+use vm_core::{
+    mast::{
+        BasicBlockNode, CallNode, DynNode, ExternalNode, JoinNode, LoopNode, MastForest, MastNode,
+        MastNodeId, OpBatch, SplitNode, OP_GROUP_SIZE,
+    },
+    stack::MIN_STACK_DEPTH,
+    utils::range,
+    Decorator, DecoratorIterator, Felt, Kernel, Operation, Program, StackOutputs, Word, EMPTY_WORD,
+    ONE, WORD_SIZE, ZERO,
+};
+
+use crate::{
+    operations::utils::assert_binary, utils::resolve_external_node, ContextId, ExecutionError,
+    Host, ProcessState, FMP_MIN, SYSCALL_FMP_MIN,
+};
+
+// temporary module to
+pub mod experiments;
+
+mod memory;
+
+// Ops
+mod crypto_ops;
+mod field_ops;
+mod fri_ops;
+mod horner_ops;
+mod io_ops;
+mod stack_ops;
+mod sys_ops;
+mod u32_ops;
+
+#[cfg(test)]
+mod tests;
+
+/// The size of the stack buffer.
+///
+/// Note: This value is much larger than it needs to be for the majority of programs. However, some
+/// existing programs need it (e.g. `std::math::secp256k1::group::gen_mul`), so we're forced to push
+/// it up. At this high a value, we're starting to see some performance degradation on benchmarks.
+/// For example, the blake3 benchmark went from 285 MHz to 250 MHz (~10% degradation). Perhaps a
+/// better solution would be to make this value much smaller (~1000), and then fallback to a `Vec`
+/// if the stack overflows.
+const STACK_BUFFER_SIZE: usize = 6650;
+
+/// The initial position of the top of the stack in the stack buffer.
+///
+/// We place this value close to 0 because if a program hits the limit, it's much more likely to hit
+/// the upper bound than the lower bound, since hitting the lower bound only occurs when you drop
+/// 0's that were generated automatically to keep the stack depth at 16. In practice, if this
+/// occurs, it is most likely a bug.
+const INITIAL_STACK_TOP_IDX: usize = 50;
+
+/// A fast processor which doesn't generate any trace.
+#[derive(Debug)]
+pub struct FastProcessor {
```

Comment:
I might have missed it, but it would be good to describe how two things are handled by the fast processor:

1. Stack management - e.g., how the bound checks work and what happens when we switch contexts.
2. Clock cycle management - e.g., how it is incremented, and how can we compute the "true" clock cycle of the VM.

Reply:
Done